pkg/executor/show.go

```diff
@@ -2112,6 +2113,46 @@ func (e *ShowExec) appendRow(row []any) {
 	}
 }
 
+func (e *ShowExec) fetchShowDistributions(ctx context.Context) error {
+	tb, err := e.getTable()
+	if err != nil {
+		return errors.Trace(err)
+	}
+	physicalIDs := []int64{}
+	if pi := tb.Meta().GetPartitionInfo(); pi != nil {
+		for _, name := range e.Table.PartitionNames {
+			pid, err := tables.FindPartitionByName(tb.Meta(), name.L)
+			if err != nil {
+				return err
+			}
+			physicalIDs = append(physicalIDs, pid)
+		}
+		if len(physicalIDs) == 0 {
+			for _, p := range pi.Definitions {
+				physicalIDs = append(physicalIDs, p.ID)
+			}
+		}
```

Comment:
Why are there two ways to obtain the ID, and what are their differences? I noticed that this is copied from the code below; it would be best to wrap the duplicate code in a function and replace similar instances elsewhere.

Reply:
It's different, `show regions` supports the index filter but the `show distribution` doesn't need. to support this.